crates/bevy_transform/src/hierarchy/child_builder.rs

```diff
@@ -219,14 +219,16 @@ impl<'w> BuildWorldChildren for EntityMut<'w> {
     fn push_children(&mut self, children: &[Entity]) -> &mut Self {
         let parent = self.id();
         {
-            // SAFE: parent entity is not modified
+            // SAFE: parent entity is not modified and its location is updated manually
             let world = unsafe { self.world_mut() };
             for child in children.iter() {
                 world
                     .entity_mut(*child)
                     // FIXME: don't erase the previous parent (see #1545)
                     .insert_bundle((Parent(parent), PreviousParent(parent)));
             }
+            // Inserting a bundle in the children entities may change the parent entity's location
```

Comment:
Maybe make it clearer that this occurs if the child was in the same archetype as `self`, which I think is the cause of this issue?